src/wrappers/pubsub.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable prefer-rest-params */
+/**
+ * @fileoverview Handlers for Pubsub instrumentation
+ */
+
+const shimmer = require('shimmer');
+const {
+    tracer,
+    tryRequire,
+} = require('epsagon');
+const { initializeEvent, finalizeEvent } = require('../runners/runner');
+const traceContext = require('../trace_context.js');
+
+const GOOGLE_CLOUD_TYPES = {
+    defaultProjectId: '{{projectId}}',
+    pubsub: {
+        name: 'pubsub',
+    },
+};
+
+const subscriber = tryRequire('@google-cloud/pubsub/build/src/subscriber');
+
+/**
+ * Handle subscriber event emitter of eventName='message'
+ * @param {Message} message received message.
+ * @param {*} originalHandler listener callback function.
+ * @param {*} requestFunctionThis request arguments.
+ */
+function pubSubSubscriberMiddleware(message, originalHandler, requestFunctionThis) {
+    try {
+        tracer.restart();
+        const functionName = originalHandler.name || 'messageHandler';
+        const { event: pubSubEvent, startTime: pubSubStartTime } = initializeEvent(
+            GOOGLE_CLOUD_TYPES.pubsub.name,
+            // eslint-disable-next-line no-underscore-dangle
+            requestFunctionThis._subscription.projectId,
+            'messagePullingListener',
+            'trigger'
+        );
+        const { event: nodeEvent, startTime: nodeStartTime } = initializeEvent(
+            'node_function', functionName, 'messageReceived', 'runner'
+        );
+        tracer.addEvent(pubSubEvent);
+        const messageData = JSON.parse(`${message.data}`);
+        const callbackResponse = { messageId: message.id, ...messageData };
+        finalizeEvent(pubSubEvent, pubSubStartTime, null, callbackResponse);
+        const { label, setError } = tracer;
+        let originalHandlerErr;
+        tracer.addRunner(nodeEvent);
+        try {
+            const returnMessage = message;
+            returnMessage.epsagon = {
+                label,
+                setError,
+            };
+            originalHandler(returnMessage, {});
```

Comment:
return?

Reply:
can't this be an async function?

Reply:
> When the EventEmitter object emits an event, all of the functions attached to that specific event are called synchronously. Any values returned by the called listeners are ignored and will be discarded.
So I don't need to return.

It can be an async function and now supported.